libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -740,8 +750,13 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
             break;
         }
 
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s Request for %sing received", msg_prefix, arm_motors_request.arm ? "arm" : "disarm");
-        arm_motors_response.result = arm_motors_request.arm ? AP::arming().arm(AP_Arming::Method::DDS) : AP::arming().disarm(AP_Arming::Method::DDS);
+        if (external_control->is_enabled()) {
```

Comment:
We should ask the dev team whether they want to put `arming` in external control and push all control handover logic into external control for things like arming, mode change, joystick, etc.

Reply:
I would also ask the Dev team, what is the best way to add this functionality to the Ground Station?